vortex-fastlanes/src/bitpacking/compute.rs

```diff
@@ -39,13 +39,13 @@ impl ScalarAtFn for BitPackedArray {
         if index >= self.len() {
             return Err(vortex_err!(OutOfBounds:index, 0, self.len()));
         }
-        if self.bit_width() == 0 {
-            let ptype = self.dtype().try_into()?;
-            match_each_integer_ptype!(&ptype, |$P| {
-                return Ok(Scalar::from(0 as $P));
-            })
+
+        if let Some(patches) = self.patches() {
+            if self.bit_width == 0 || patches.is_valid(index) {
```

Comment:
If the patch value is null this is wrong?

Reply:
For posterity, we (for now) just assert that in sparse arrays, all non-null values in a patch array are patches (and vice versa). 

https://github.com/fulcrum-so/vortex/pull/194/files#diff-7021320e8750a824cf1f4153a02a9c3f38a3be9e5113138aa514063f9969fe31R22